daemon/cmd/daemon.go

```diff
@@ -581,6 +582,22 @@ func NewDaemon(ctx context.Context, epMgr *endpointmanager.EndpointManager, dp d
 		option.Config.Devices = []string{device}
 	}
 
+	if option.Config.EnableLocalRedirectPolicy {
+		found := false
+		if h := probes.NewProbeManager().GetHelpers("cgroup_sock_addr"); h != nil {
+			if _, ok := h["bpf_sk_lookup_tcp"]; ok {
+				if _, ok = h["bpf_sk_lookup_udp"]; ok {
+					found = true
+				}
+			}
+		}
+		if !found {
+			log.Warn("Without socket lookup kernel functionality, BPF " +
+				"datapath cannot prevent potential loop caused by local-redirect" +
+				"service translation. Needs kernel version >= 5.1")
```

Comment:
I wonder whether we should move this warning under a `sync.Once()` when adding an LRP. That way, even if you enable LRP, you will only be warned when using the feature. It's possible in a future release we will enable this flag by default, in which case it'd be nice to not automatically warn if the user is running a common kernel like 4.19 (which supports LRP but has this caveat).